lms/djangoapps/learner_dashboard/views.py

```diff
@@ -45,3 +45,36 @@ def view_programs(request):
     }
 
     return render_to_response('learner_dashboard/programs.html', context)
+
+
+def view_program_details(request, programId):
+    """View programs in which the user is engaged."""
+    show_program_listing = ProgramsApiConfig.current().show_program_listing
```

Comment:
It looks like you copied the body of this view from `view_programs()` above. For the time being, it should be sufficient to use the existing config model to determine whether to pass a skeleton context to the template or 404. I think you'll want to add a new config field for controlling display of the detail view. Let me or @schenedx know if you need help with migrations.